streamz/core.py

```diff
@@ -497,6 +534,28 @@ def to_batch(self, **kwargs):
         from .batch import Batch
         return Batch(stream=self, **kwargs)
 
+    def _retain_refs(self, metadata):
+        if not isinstance(metadata, list):
+            metadata = [metadata]
+        for m in metadata:
+            if m and 'refs' in m:
```

Comment:
Is `m` ever `None`? 
How about `refs = m.get('refs', []`)`
Also, no need to create the list every time
```
if isinstance(refs, list):
    for ref in refs:
        ref.retain(1)
else:
    refs.retain(1)
```
Note that the value of 1 is already the default value.

Reply:
This code has been removed and simplified.